core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/db/LegacyStructuredEventDBService.java

```diff
@@ -158,16 +165,24 @@ public StructuredEventContainer getStructuredEventsForStackByCrn(String crn, Lon
         return getEventsForUserWithResourceId(stack.getType().getResourceType(), stack.getId());
     }
 
+    public void deleteEntriesByResourceIdsOlderThanThreeMonths(Long resourceIds) {
+        repository.deleteRecordsOlderThan(resourceIds, getTimestampThatThreeMonthsBeforeNow());
+    }
+
+    public void deleteEntriesForAccountThatIsOlderThanThreeMonths(String accountId) {
```

Comment:
Isn't this logic will delete all the structured events which are older than 3 months even for clusters that are running?
Is it expected?
cc: @akanto

Reply:
yes it will. having a requirement confirmation from Attila would be nice 🙂

Reply:
@gregito  with @akanto we discussed the issue and please delete only cluster related structured events for clusters that's already been terminated and only events which are older than 3 months.
Could you please update the mentioned logic?